### ISSUE TITLE
Cannot recognized option "suggest-join-with-non-empty-separator"

### ISSUE THREAD

**Issue by winstonsung** [pylint-dev/pylint#9836]
### Bug description

```python
"message": "Unrecognized option found: suggest-join-with-non-empty-separator",
"message-id": "E0015"
```
```


### Configuration

```ini
(...)

[REFACTORING]

(...)

# Let 'consider-using-join' be raised when the separator to join on would be
# non-empty (resulting in expected fixes of the type: ``"- " + " -
# ".join(items)``)
suggest-join-with-non-empty-separator=yes

[REPORTS]

(...)

```
```


### Command used

```shell
Visual Studio Code


(...)\Python311\python.exe -m pylint --reports=n --output-format=json --clear-cache-post-run=y --from-stdin (...).py
```
```


### Pylint output

```python
[
    {
        "type": "error",
        "module": "(...)\\.pylintrc",
        "obj": "",
        "line": 1,
        "column": 0,
        "endLine": null,
        "endColumn": null,
        "path": "(...)\\.pylintrc",
        "symbol": "unrecognized-option",
        "message": "Unrecognized option found: suggest-join-with-non-empty-separator",
        "message-id": "E0015"
    }
]
```

### Expected behavior

Should either be identified or removed from the default generated `.pylintrc` file + address in release notes.

### Pylint version

```shell
pylint 3.2.6
astroid 3.2.4
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
```

### OS / Environment

Windows 10, version 22H2, build 19045.4651
Visual Studio Code 1.91.1

**Comment by DanielNoord** [comment]
U don't really understand your issue. This is an option and the branch on which you added a commit to make this into a comment passed CI successfully before. Could it be that you were running an old version locally?

**Comment by winstonsung** [comment]
Yeah, it didn't affect the CI process, but it's showing the error on the Visual Studio Code on my Windows laptop.

The listed environments should always explained that there's no issue on GitHub Actions but having issues with Visual Studio Code with the latest version.

**Comment by DanielNoord** [comment]
That wasn't clear to me from the post. Still you should probably upgrade the version of your extension or make it use an update local version. This is not an issue that can be solved in this repository, there is a guide online on how to update the version in the VS Code plugin.

**Comment by winstonsung** [comment]
> upgrade the version of your extension

I'm not sure which VS Code extension are you referring to. The Pylint I used is installed from `pip install` .

> make it use an update local version

I believe pylint 3.2.6 is already the latest version currently?

**Comment by DanielNoord** [comment]
Please see the plugin page:
https://marketplace.visualstudio.com/items?itemName=ms-python.pylint

The version of `pylint` that is bundled with it does not support this option (yet). You can ask the maintainers of the plugin to release a new version which bumps the bundled version or make it point to your location installation of `pylint` instead of using the bundled version.

**Comment by winstonsung** [comment]
Oh, great thanks for the explanation.
